apps/coscrad-frontend/src/components/Notes/NoteIndex.container.tsx

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { useAppDispatch } from '../../app/hooks';
+import { RootState } from '../../store';
+import { fetchNotes } from '../../store/slices/noteSlice/thunks';
+import { ErrorDisplay } from '../ErrorDisplay/ErrorDisplay';
+import { Loading } from '../Loading';
+import { NoteIndexPresenter } from './NoteIndex.presenter';
+
+export const NoteIndexContainer = (): JSX.Element => {
+    const { data: notes, isLoading, errorInfo } = useSelector((state: RootState) => state.notes);
+
+    const dispatch = useAppDispatch();
+
+    useEffect(() => {
+        if (notes === null) dispatch(fetchNotes());
```

Comment:
The thunk and the presenter seem to be the only 2 pieces of information that are specific to the aggregate type.